Initialize keymap of virtual keyboard (needed if no GUI displayed before)

Signed-off-by: Laurent Merckx <laurent-merckx@skynet.be>

diff --git a/backends/vkeybd/virtual-keyboard.h b/backends/vkeybd/virtual-keyboard.h
index 38139e2ad8..6be4c31af8 100644
--- a/backends/vkeybd/virtual-keyboard.h
+++ b/backends/vkeybd/virtual-keyboard.h
@@ -242,6 +242,7 @@ protected:
 	bool checkModeResolutions();
 	void switchMode(Mode *newMode);
 	void switchMode(const String &newMode);
+	void initKeymap();
 	void handleMouseDown(int16 x, int16 y);
 	void handleMouseUp(int16 x, int16 y);
 	String findArea(int16 x, int16 y);
diff --git a/backends/vkeybd/virtual-keyboard.cpp b/backends/vkeybd/virtual-keyboard.cpp
index 74cf1f2631..e7a57ee9a9 100644
--- a/backends/vkeybd/virtual-keyboard.cpp
+++ b/backends/vkeybd/virtual-keyboard.cpp
@@ -24,6 +24,7 @@
 
 #ifdef ENABLE_VKEYBD
 
+#include "gui/gui-manager.h"
 #include "backends/vkeybd/virtual-keyboard.h"
 
 #include "backends/keymapper/keymapper.h"
@@ -223,6 +225,21 @@ void VirtualKeyboard::handleMouseUp(int16 x, int16 y) {
 	_kbdGUI->endDrag();
 }
 
+// If no GUI opened before the virtual keyboard, kKeymapTypeGui is not yet initialized
+// Check and do it if needed
+void VirtualKeyboard::initKeymap() {
+        using namespace Common;
+
+        Keymapper *mapper = _system->getEventManager()->getKeymapper();
+
+        // Do not try to recreate same keymap over again
+        if (mapper->getKeymap(kGuiKeymapName) != 0)
+                return;
+
+        Keymap *guiMap = g_gui.getKeymap();
+        mapper->addGlobalKeymap(guiMap);
+}
+
 void VirtualKeyboard::show() {
 	if (!_loaded) {
 		debug(1, "VirtualKeyboard::show() - Virtual keyboard not loaded");
@@ -234,6 +251,7 @@ void VirtualKeyboard::show() {
 	switchMode(_initialMode);
 
 	{
+		initKeymap();
 		KeymapTypeEnabler guiKeymap(_system->getEventManager()->getKeymapper(), Keymap::kKeymapTypeGui);
 		_kbdGUI->run();
 	}
